File: 3731-find-missing-elements/3731-find-missing-elements.cpp
class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        vector<int> res;
        vector<bool> range(101, false);
        char mn = 100;
        char mx = 1;
        for(char i = 0; i < nums.size(); i++)
        {
            mn = min((int)mn, nums[i]);
            mx = max((int)mx, nums[i]);
            range[nums[i]] = true;
        }
        for(char i = mn + 1; i < mx; i++)
        {
            if(!range[i])
            {
                res.push_back(i);
            }
        }
        return res;
    }
};

File: 3731-find-missing-elements/3731-find-missing-elements.py
class Solution:
    def findMissingElements(self, nums: List[int]) -> List[int]:
        res = []
        rng = [False for _ in range(101)]
        mn = 100
        mx = 1
        for num in nums:
            mn = min(mn, num)
            mx = max(mx, num)
            rng[num] = True
        for i in range(mn + 1, mx):
            if not rng[i]:
                res.append(i)
        return res
